decidim/decidim
# Open and close the Bulletin Board ballot box for elections

#### :tophat: What? Why?
This PR implements the open and close ballot box for elections. 

First, it applies the changes in the gem to the setup election command and converts the setup election page into a steps page.

Within that steps page, it adds the ready and vote steps with the open and close buttons.

Finally, it adds a new rake task to be scheduled to automatize the opening and the closing of the ballot box. The ballot box will be automatically opened when running the task a few hours before the starting of the election ([open_ballot_box_maximum_hours_before_start configuration](https://github.com/decidim/decidim/pull/7035/files#diff-aa57d4b4daa20683e6915506ff7a3229ad3826ad02b012d0e06ca24aad839b63R24)) and automatically closed when running the task after the election ending time.

#### :pushpin: Related Issues
- Fixes #6632
- Fixes #6634

#### Testing
* Create and setup an election to be started in 3 hours and to be finished 10 minutes after that.
* Perform the key ceremony process.
* Go to the steps page of the election.
![imagen](https://user-images.githubusercontent.com/453545/102722511-d30ffa00-4301-11eb-93cd-06121e35d01f.png)
* Press the "Open ballot box" button.
![imagen](https://user-images.githubusercontent.com/453545/102722312-a0193680-4300-11eb-97ba-2afe098eb96e.png)
* Wait the ending time of the election.
* Go to the steps page of the election.
* Press the "Close ballot box" button.
![imagen](https://user-images.githubusercontent.com/453545/102757234-4dc42e00-4371-11eb-824a-b100c0184029.png)
* The election should be in the "Tally" status now.
  <img width="1094" alt="Screenshot 2020-12-23 at 11 36 54" src="https://user-images.githubusercontent.com/210216/102987822-39b83200-4513-11eb-8c56-ccf785e5c393.png">


#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.adoc) to this repository.


### :camera: Screenshots
:hearts: Thank you!
